components/scream/src/control/surface_coupling.cpp

```diff
@@ -268,43 +364,57 @@ get_col_info(const std::shared_ptr<const FieldHeader>& fh,
   const auto& dims = layout.dims();
 
   auto lt = get_layout_type(layout.tags());
-  const bool scalar = lt==LayoutType::Scalar2D || lt==LayoutType::Scalar3D;
-  const bool vector = lt==LayoutType::Vector2D || lt==LayoutType::Vector3D;
+  const bool scalar   = lt==LayoutType::Scalar2D || lt==LayoutType::Scalar3D;
+  const bool vector   = lt==LayoutType::Vector2D || lt==LayoutType::Vector3D;
+  const bool layout3d = lt==LayoutType::Scalar3D || lt==LayoutType::Vector3D;
   EKAT_REQUIRE_MSG(scalar || vector,
       "Error! Support for tensor fields not yet implemented in surface coupling.\n");
   EKAT_REQUIRE_MSG( (vecComp<0) == scalar,
       "Error! You can and must specify a vector component only for vector fields.\n");
 
   vecComp = std::max(0,vecComp);
 
-  // Product of dimensions of field, without counting space dims, that is,
-  // the product of dimensions of the "analytical" field.
-  int dims_prod = 1;
-
-  // Compute initial offset based on this field rank and requested component
+  // Compute initial offset. We use get_last_extent() as it
+  // accounts for padding (if any).
   col_offset = vecComp;
-  if (vector) {
-    dims_prod = dims[1];
+  if (layout3d) {
+    if (lt==LayoutType::Vector3D) {
+      col_offset *= fh->get_alloc_properties().get_last_extent();;
+    }
+    col_offset += m_num_levs-1;
```

Comment:
I am not sure if we ever export/import from/into an interface field. Maybe @AaronDonahue can tell us. However, we could just use `fh->get_identifier().get_layout().dims().back()`. This assumes that the number of levels is the last dimension, but I think that's safe. Or, alternatively, be super safe and do:
```
const auto& layout = fh->get_identifier().get_layout();
const bool at_midpoints = layout.has_tag(VL);
const int num_levs = at_midpoints ? m_num_levs : m_num_levs+1;
col_offset += num_levs-1;
```

Reply:
Ahh, forgot about this as well.

Reply:
Actually never mind, I had included it.